src/app/components/main-search-box/main-search-box.component.html

```diff
@@ -0,0 +1,174 @@
+            <div class="airport-code">{{ airport[1] }}</div>
+          </div>
+        </mat-option>
+      </mat-select>
+      <mat-error *ngIf="searchFlyForm.controls['from'].hasError('required')">
+        Please enter a departure
+      </mat-error>
+    </mat-form-field>
+
+    <mat-form-field class="dest-input" appearance="outline">
+      <mat-label>Destination</mat-label>
+      <mat-select class="select" formControlName="destination">
+        <mat-option *ngFor="let airport of airports" class="option" [value]="airport[1]">
+          <div class="airport-select">
+            <div class="airport-name">{{ airport[0] }}</div>
+            <div class="airport-code">{{ airport[1] }}</div>
+          </div>
+        </mat-option>
+      </mat-select>
+      <mat-error *ngIf="searchFlyForm.controls['destination'].hasError('required')">
+        Please enter a destination
+      </mat-error>
+    </mat-form-field>
+  </div>
+  <div class="date-wrapper">
+    <mat-form-field
+      class="date-picker"
+      *ngIf="roundTrip; else oneWayDatePicker"
+      appearance="outline"
+    >
+      <mat-label>Choose a dates</mat-label>
+      <mat-date-range-input [rangePicker]="picker">
+        <input matStartDate placeholder="Start date" formControlName="departureDate" />
+        <input matEndDate placeholder="End date" formControlName="departureReturnDate" />
+      </mat-date-range-input>
+      <mat-error *ngIf="searchFlyForm.controls['departureDate'].hasError('required')">
+        Please enter a departure date
+      </mat-error>
+      <mat-error *ngIf="searchFlyForm.controls['departureReturnDate'].hasError('required')">
+        Please enter a return date
+      </mat-error>
+      <mat-datepicker-toggle matIconSuffix [for]="picker"></mat-datepicker-toggle>
+      <mat-date-range-picker #picker></mat-date-range-picker>
+    </mat-form-field>
+
+    <ng-template #oneWayDatePicker>
+      <mat-form-field class="date-picker" appearance="outline">
+        <mat-label>Choose a date</mat-label>
+        <input matInput [matDatepicker]="picker" formControlName="departureDate" />
+        <mat-error *ngIf="searchFlyForm.controls['departureDate'].hasError('required')">
+          Please enter a departure date
+        </mat-error>
+        <mat-datepicker-toggle matIconSuffix [for]="picker"></mat-datepicker-toggle>
+        <mat-datepicker #picker></mat-datepicker>
+      </mat-form-field>
+    </ng-template>
+
+    <mat-form-field class="passengers-input" appearance="outline">
+      <mat-label>Passengers</mat-label>
+      <mat-select class="select">
+        <mat-select-trigger>
+          <span *ngIf="passengers.adults > 0"> Adults {{ passengers.adults }} </span>
+          <span *ngIf="passengers.children > 0"> Children {{ passengers.children }} </span>
+          <span *ngIf="passengers.infants > 0"> Infants {{ passengers.infants }} </span>
+        </mat-select-trigger>
+
+        <mat-option value="adult">
```

Comment:
This option template repeats 3 times - for adult, child and infant. 
I suggest you create separate component and wrap this `<mat-option>`. 
Then, you will just need pass simple config with labels ("Adult", "14+ years", ...) and use `@Output` emitters to notify parent form about changes in counters. 

So the final template will look like this:
```
<app-passengers-option 
 [config]="adultsPassengersConfig" 
 (onChange)="handlePassengersChange('adult', $event)">
```